cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/program/MessagingProgramStateWriter.java

```diff
@@ -56,7 +58,8 @@
 
 
   @Inject
-  public MessagingProgramStateWriter(CConfiguration cConf, MessagingService messagingService) {
+  public MessagingProgramStateWriter(CConfiguration cConf,
+                                     @Named(MessagingClientModule.LOCAL_TMS) MessagingService messagingService) {
```

Comment:
LOCAL_TMS is misleading since this won't always be local. Would be better to name it PROGRAM_MESSAGING or something more related to this class rather than the implementation of the MessagingService.

Reply:
fixed